src/main/kotlin/dev/arunvelsriram/ktlint/rulesets/junit/MaximumIgnoredTestRule.kt

```diff
@@ -0,0 +1,35 @@
+package dev.arunvelsriram.ktlint.rulesets.junit
+
+import com.pinterest.ktlint.core.Rule
+import org.jetbrains.kotlin.com.intellij.lang.ASTNode
+import org.jetbrains.kotlin.psi.KtFunction
+import org.jetbrains.kotlin.psi.stubs.elements.KtStubElementTypes
+
+class MaximumIgnoredTestRule : Rule("maximum-ignored-test") {
+
+    override fun visit(
+        node: ASTNode,
+        autoCorrect: Boolean,
+        emit: (offset: Int, errorMessage: String, canBeAutoCorrected: Boolean) -> Unit
+    ) {
+        if (ignoredTestsCount > IGNORED_TESTS_THRESHOLD) return
+        if (node.elementType == KtStubElementTypes.FUNCTION) {
+            val func = node.psi as KtFunction
+            func.annotationEntries.find {
+                it.text == IGNORE_ANNOTATION
+            }?.let {
+                if (ignoredTestsCount == IGNORED_TESTS_THRESHOLD) {
+                    emit(0, LINT_ERROR_MESSAGE, false)
+                }
+                ignoredTestsCount += 1
```

Comment:
we could swap 24 and 21 to 23 so that emit happens immediately after count becomes 5 instead of the next visit.

Reply:
If the codebase has `5` errors, there should not be any lint error as the threshold is `5` . An error should be thrown when the 6th ignored test is encountered.

Reply:
Yes. Sorry my bad.